module/data/shared/damage-field.mjs

```diff
@@ -0,0 +1,126 @@
+import FormulaField from "../fields/formula-field.mjs";
+
+const { BooleanField, EmbeddedDataField, NumberField, SchemaField, SetField, StringField } = foundry.data.fields;
+
+/**
+ * Field for storing damage data.
+ */
+export default class DamageField extends EmbeddedDataField {
+  constructor(options) {
+    super(DamageData, options);
+  }
+}
+
+/* -------------------------------------------- */
+
+/**
+ * Data model that stores information on a single damage part.
+ *
+ * @property {number} number           Number of dice to roll.
+ * @property {number} denomination     Die denomination to roll.
+ * @property {string} bonus            Bonus added to the damage.
```

Comment:
This is fine so long as we're absolutely clear it's just for ease-of-input purposes, and that we are only ever using `DamageData#formula` everywhere else in the system.